### PR TITLE
Add more array scalar functions

### PR DESCRIPTION
**Pull request by wirybeaver**
Implement array scalar functions raised in the https://github.com/apache/pinot/issues/11442

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java [hunk 1]
```diff
@@ -74,6 +76,50 @@ public static int arrayIndexOfString(String[] values, String valueToFind) {
     return ArrayUtils.indexOf(values, valueToFind);
   }
 
+  @ScalarFunction
+  public static int[] arrayIndexOfAllInt(int[] value, int valueToFind) {
```
Comment (xiangfu0) [inline pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java:80]: do we need to support more type like long/float/double etc?

Comment (wirybeaver) [reply pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java:80]: Will do.

File: pinot-core/src/test/java/org/apache/pinot/core/operator/transform/function/ScalarTransformFunctionWrapperTest.java [hunk 2]
```diff
@@ -958,6 +958,46 @@ public void testArrayElementAtString() {
     testTransformFunction(transformFunction, expectedValues);
   }
 
+  @Test
+  public void testArrayIndexOfAllString() {
+    ExpressionContext expression = RequestContextUtils.getExpression(
+        String.format("array_index_of_all_string(%s, 'a')", STRING_ALPHANUM_MV_COLUMN_2));
```
Comment (xiangfu0) [inline pinot-core/src/test/java/org/apache/pinot/core/operator/transform/function/ScalarTransformFunctionWrapperTest.java:964]: Add a test for `array_index_of_all_int`

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java [hunk 3]
```diff
@@ -74,6 +76,85 @@ public static int arrayIndexOfString(String[] values, String valueToFind) {
...
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java:80]: Consider naming it `arrayIndexesOfInt()` (align with the method name in `ArrayUtils`), same for other methods

File: pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java [hunk 4]
```diff
@@ -74,6 +76,85 @@ public static int arrayIndexOfString(String[] values, String valueToFind) {
...
+    List<Integer> indices = new ArrayList<>();
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/function/scalar/ArrayFunctions.java:81]: Let's use `IntArrayList` from `it.unimi.dsi.fastutil.ints`, then we can use `list.toIntArray()`

**Review by Jackie-Jiang — APPROVED** [review]
LGTM in general

**Comment by Jackie-Jiang** [comment]
@xiangfu0 there are still pending comments

@wirybeaver Can you address the pending comments in a separate PR?

**Comment by wirybeaver** [comment]
Sorry for the late reply, should have solved the comment. I will rename arrayIndexOfAll to arrayIndexesOf and use the it.unimi.dsi.fastutil.ints

**Pull request merged**